app/src/androidTest/java/ch/epfl/sdp/appart/UserProfileActivityTest.java

```diff
@@ -60,19 +74,29 @@
     @Rule(order = 1)
     public ActivityScenarioRule<MainActivity> mActivityTestRule = new ActivityScenarioRule<>(MainActivity.class);
 
-    @BindValue
-    LoginService login = new MockLoginService();
+    /* Used to grant camera permission always */
+    //@Rule
+    //public GrantPermissionRule permissionRule = GrantPermissionRule.grant(Manifest.permission.CAMERA);
+
+    /* Used to grant camera permission always */
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule =
+            GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION);
 
     @BindValue
     DatabaseService database = new MockDatabaseService();
 
+    @BindValue
+    LoginService login = new MockLoginService();
+
     @Before
     public void init() {
         Intents.init();
         hiltRule.inject();
     }
     @Test
-    public void userProfileActivityTest() {
+    public void userProfileActivityTest() throws UiObjectNotFoundException {
```

Comment:
I don't really understand what this tests. Also, it is not very readable, the variable names are not really explicit.

Reply:
Thank you @ADGLY for you review! Indeed good point for the complete exceptionally, I added them where needed. 

This test was done with espresso record testing (which is the reason why the variable names are not explicit). 

### Test explained
The test is a single test which tries all the actions that may be performed by the user and then checks whether each of these actions took place correctly. The specifics on the button instantiations are checks.

**this test does **not** test camera activity but UserProfileActivity functions on camera activity result.**

**In the order in which its executed:**

1. Edit **name**, **age**, **phone number** and **gender**.
2. Press on **change profile image button** - this starts the CamerActivity intent in UserProfileActivity with "onActivityResult".
3. The test creates a mock response with empty uri from the camera activity to the UserProfileActivity (the actual camera is never called) .
4. The **DONE** button is pressed and all the information (name, age, etc..) and the new picture path is saved in MockDatabase.
5. The **edit profile** is pressed again, the remove button is pressed (the image is removed from MockDatabase).
6. The fields are checked for right text.
7. The test terminates.

Reply:
Oh ok, do you check that the CameraActivity is started in step 2 ? Also, do you check that the new picture path is the correct one ? Sorry, for all the questions but I want to make sure I understand the test.

Reply:
Yes good point about the path checks! I added tests to make sure the camera activity is actually called and to verify the picture paths and the adding and removing of picture path in MockDatabase